src/context.rs

```diff
@@ -82,82 +82,57 @@ pub trait AsBaseAudioContext {
-    /// `.wav` files and do not perform any resampling operation. Be carefull
-    /// to use audio files that match the current sample rate or the file will
-    /// pitched when played.
-    fn decode_audio_data(&self, file: std::fs::File) -> AudioBuffer {
-        // @see - https://github.com/orottier/web-audio-api-rs/issues/42
-        // @note - should also be async, but that's not a big deal neither for now
-        let buf_reader = std::io::BufReader::new(file);
-        let mut reader = hound::WavReader::new(buf_reader).unwrap();
-        let hound::WavSpec {
-            channels,
-            sample_rate: _, // @todo - use for resampling
-            bits_per_sample,
-            sample_format,
-        } = reader.spec();
-
-        let number_of_channels = channels as usize;
-        let length = reader.duration() as usize;
-
-        let mut decoded: Vec<Vec<f32>> = Vec::with_capacity(number_of_channels);
-
-        for _ in 0..number_of_channels {
-            decoded.push(Vec::<f32>::with_capacity(length));
-        }
-
-        // @note - hound retrieves interleaved values
-        // cf. https://docs.rs/hound/latest/hound/struct.WavReader.html#method.samples
-        match sample_format {
-            hound::SampleFormat::Int => {
-                let mut channel_number = 0;
-
-                if bits_per_sample == 16 {
-                    for sample in reader.samples::<i16>() {
-                        let s = f32::from(sample.unwrap()) / f32::from(i16::MAX);
-                        assert!(s.abs() <= 1.);
-                        decoded[channel_number].push(s);
-                        // next sample belongs to the next channel
-                        channel_number = (channel_number + 1) % number_of_channels;
-                    }
-                } else {
-                    panic!("bits_per_sample {:?} not implemented", bits_per_sample);
-                }
-            }
-            hound::SampleFormat::Float => {
-                // this one is not tested
-                let mut channel_number = 0;
-
-                for sample in reader.samples::<f32>() {
-                    let s = sample.unwrap();
-                    decoded[channel_number].push(s);
-                    // next sample belongs to the next channel
-                    channel_number = (channel_number + 1) % number_of_channels;
-                }
-            }
-        }
-
-        assert_eq!(decoded[0].len(), length);
-
-        // [spec] Take the result, representing the decoded linear PCM audio data,
-        // and resample it to the sample-rate of the BaseAudioContext if it is
-        // different from the sample-rate of audioData.
-
-        // @todo - resample if needed
-        // if (sample_rate != self.sample_rate()) {}
-
-        let mut channels = Vec::<ChannelData>::with_capacity(number_of_channels);
+    /// The current implementation can decode FLAC, Opus, PCM, Vorbis, and Wav.
```

Comment:
Do you think it could support `mp3` as well?

Reply:
Definitely, Symphonia supports quite a few more types: https://docs.rs/symphonia/0.4.0/symphonia/#codecs
The current ones are the 'default features', I'll add the mp3-feature too because it's nice to have batteries included. For the others we can also include feature gates as the added code size may not be beneficial for these lesser used codecs

Reply:
Cool!